lib/Sema/DerivedConformanceDifferentiable.cpp

```diff
@@ -625,6 +634,10 @@ getOrSynthesizeSingleAssociatedStruct(DerivedConformance &derived,
                               None))
       inherited.push_back(kpIterableType);
   }
+  // If all members conform to `ElementaryFunctions`, make the associated struct
```

Comment:
This logic (conform synthesized `Differentiable` associated types to `ElementaryFunctions` *if possible*) is not super principled. Similar logic exists for conforming synthesized types to `VectorProtocol` *if possible*.

We could remove this logic by constraining `TangentVector` to `VectorProtocol` and/or `ElementaryFunctions`, but it's not clear to me whether that's desirable. Constraining `TangentVector` to `VectorProtocol` seems more reasonable.